lib/Parse/ParseExpr.cpp

```diff
@@ -3041,7 +3050,24 @@ ExprResult Parser::ParseBoundsCastExpression(IdentifierInfo &Ident,
   // if there is rel, parsing it.
   if (Tok.is(tok::comma)) {
     ConsumeToken();
-    //...
+    if (StartsRelativeBoundsClause(Tok)) {
+      IdentifierInfo *Ident = Tok.getIdentifierInfo();
+      bool isError = false;
+      ConsumeToken();
+      BalancedDelimiterTracker PT(*this, tok::l_paren);
+      if (PT.expectAndConsume(diag::err_expected_lparen_after,
+                              Ident->getNameStart())) {
+        SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
+      } else {
+        RelativeClause =
+            CreateRelativeBoundsClause(isError, Ident, Tok.getLocation());
+        if (isError)
+          SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
+      }
+      PT.consumeClose();
+    } else {
```

Comment:
Generally the clang code avoid putting braces around single statements in else branches.   The code elsewhere has the from:
```
else 
   SkipUntil (tok::greater, ...
```